hikari/impl/bot.py

```diff
@@ -758,30 +761,33 @@ def die() -> None:
             finally:
                 loop.run_until_complete(self.join())
         except errors.GatewayClientClosedError as ex:
-            _LOGGER.info(str(ex))
+            _LOGGER.info("client closed with reason: %s", ex)
         finally:
             for signum in kill_signals:
                 # Windows is dumb and doesn't support signals properly.
                 with contextlib.suppress(NotImplementedError):
                     loop.remove_signal_handler(signum)
 
             if finalize_loop_on_close:
-                _LOGGER.debug("closing asyncgens for event loop %s", loop)
-                loop.run_until_complete(loop.shutdown_asyncgens())
+                remaining_tasks = [t for t in asyncio.all_tasks(loop) if not t.done()]
```

Comment:
2 questions here, 1 isn't asyncio.all_tasks meant to only return not yet finished tasks anyway and will this ever return a shielded task?

Reply:
1. implementation dependent.
2. cancelling the task will do nothing, so it will just cause it to wait until it has completed on line 779:
`loop.run_until_complete(asyncio.gather(*remaining_tasks, return_exceptions=True))`